src/handlers/rendered_link.rs

```diff
@@ -25,32 +35,47 @@ pub async fn handle(ctx: &Context, event: &Event) -> anyhow::Result<()> {
 }
 
 async fn add_rendered_link(ctx: &Context, e: &IssuesEvent, prefix: &str) -> anyhow::Result<()> {
-    if e.action == IssuesAction::Opened {
+    if e.action == IssuesAction::Opened
+        || e.action == IssuesAction::Closed
+        || e.action == IssuesAction::Reopened
+    {
+        let edit = EditIssueBody::new(&e.issue, "RENDERED_LINK");
```

Comment:
I'm a little concerned about the approach of using `EditIssueBody` for a few reasons:

* AFAIK, the triagebot comments persist in the git log.
* Some users manually add their own Rendered links, which I think will conflict with this (it seems it will add multiple rendered links).
* This may conflict with all RFCs that are currently open.

Would it be possible to keep the same logic of just manually looking for the link?

Reply:
> * AFAIK, the triagebot comments persist in the git log.

I'm sure I understand this point. Are you saying that triagebot comments are somehow included in git log?

> * Some users manually add their own Rendered links, which I think will conflict with this (it seems it will add multiple rendered links).

I don't think it would as triagebot will add the comment and surround it with hidden html marker (which is the reason I wanted to use `EditIssueBody`, since those markers simplifies many things).

https://github.com/rust-lang/triagebot/blob/f563bc9578ae26d825f0ab6b1255068afebe6a2a/src/interactions.rs#L100-L106

> * This may conflict with all RFCs that are currently open.

Good point. I can add a defensive mechanism for older RFCs.

> Would it be possible to keep the same logic of just manually looking for the link?

The issue is not really looking for the link but updating it as well, which is heavily simplified by `EditIssueBody`.

Reply:
> Are you saying that triagebot comments are somehow included in git log?

Yea. When a PR is merged, GitHub sets the contents of the PR description body as the commit message. AFAIK it does not strip HTML comments.

> I don't think it would as triagebot will add the comment and surround it with hidden html marker (which is the reason I wanted to use `EditIssueBody`, since those markers simplifies many things).

If someone posts a PR, and they manually add a rendered link, the body will look like:

```
Hey, here's my new PR!

[Rendered](https://github.com/octocat/rfcs/blob/4917eb9a8c2a3da10a5e46b080498b33655b1054/text/0001-foo.md)
```

and then triagebot would edit it to be:

```
Hey, here's my new PR!

[Rendered](https://github.com/octocat/rfcs/blob/4917eb9a8c2a3da10a5e46b080498b33655b1054/text/0001-foo.md)

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_RENDERED_LINK_START -->
[Rendered](https://github.com/octocat/rfcs/blob/branch/text/0001-foo.md)

<!-- TRIAGEBOT_RENDERED_LINK_DATA_START$${"rendered_link":"https://github.com/rust-lang/rfcs/blob/branch/text/0001-foo.md"}$$TRIAGEBOT_RENDERED_LINK_DATA_END -->

<!-- TRIAGEBOT_RENDERED_LINK_END -->
<!-- TRIAGEBOT_END -->
```

This will render in Markdown as two rendered links.

> The issue is not really looking for the link but updating it as well, which is heavily simplified by `EditIssueBody`.

I understand that `EditIssueBody` makes it easier to find and edit content, but it does not work well with PRs (and thus why we never use it there).

Reply:
Alright, changed to a more manual approach.